src/fpmath.h

```diff
@@ -41,30 +41,23 @@
 #include "powerpc_fpmath.h"
 #endif
 
-#if defined(__BYTE_ORDER__) && defined(__ORDER_LITTLE_ENDIAN__)
-
 /* Definitions provided directly by GCC and Clang. */
-#define _LITTLE_ENDIAN    __ORDER_LITTLE_ENDIAN__
-#define _BIG_ENDIAN       __ORDER_BIG_ENDIAN__
-#define _PDP_ENDIAN       __ORDER_PDP_ENDIAN__
-#define _BYTE_ORDER       __BYTE_ORDER__
+#if !(defined(__BYTE_ORDER__) && defined(__ORDER_LITTLE_ENDIAN__) && defined(__ORDER_BIG_ENDIAN__))
 
-#elif defined(__GLIBC__)
+#if defined(__GLIBC__)
```

Comment:
Instead of leaving all of this code there, would it be worth just throwing it away and seeing what breaks? It may be the case that Windows is the only platform that doesn't define these. All the other systems use a compiler that provides support for these definitions.